augur/filter.py

```diff
@@ -485,6 +527,19 @@ def run(args):
             print("ERROR: All samples have been dropped! Check filter rules and metadata file format.", file=sys.stderr)
             return 1
 
+    if args.output_metadata:
+        metadata_df = pd.DataFrame([meta_dict[strain] for strain in seq_keep])
+        metadata_df.to_csv(
+            args.output_metadata,
+            sep="\t",
+            index=False
+        )
+
+    if args.output_strains:
+        with open(args.output_strains, "w") as oh:
```

Comment:
In the case where `sequences_written=0` we don't get to this stage in the code, which means that an existing `args.output_strains` file may remain which could be confusing to the user. Ideally, that file would be cleared in this case.  Ditto for metadata output.

Reply:
Thank you for catching this! This bug was a side effect of returning 1 when no sequences were kept before the code to write out metadata or strains. I reorganized this code in c49a0ec such that the error gets thrown at the end of the filter after writing out empty strains, metadata, and sequences and writing out the report about which strains got filtered. This reshuffling also closes #681.